src/components/login/profile.tsx

```diff
@@ -15,14 +18,17 @@ const Profile = () => {
     await authService.logout();
     setUser(null);
   };
+
+  const back = () => {
+    const lastUrl = sessionStorage.getItem("lastUrl");
+    lastUrl ? (window.location.href = lastUrl) : navigate("/");
+  }
   
   let content;
   if (user === undefined)
     content = "Loading user data...";
   else if (user === null) 
-    content = <div style={{margin: "2em 0"}}>
-        <Alert variant="danger">You are not logged in.</Alert>
-      </div>;
+    back()
```

Comment:
Suggestion:

```ts
  else if (user === null) {
    content = "Not logged in!";
    back();
  } 
```

Not leaving content undefined is a bit cleaner, and this will be visible when the browser is busy and the async redirection does not happen immediately. Normally, the user does not see this, so this doesn't need to be styled like before,

Reply:
Once we have more routes that should not be entered by unauthenticated users, we should think about adding some more general mechanism like guarded routes, or simply routes with a loader function that prefetches the user, and does not even navigate to the route if this fails. The component will then also not need to fetch the user itself like here with `useEffect`.

Reply:
Session cookies are likely to be the less cumbersome way of going about it, along with protected routes.